packages/plugins/dart/flutter-freezed/src/config.ts

```diff
@@ -0,0 +1,522 @@
+   * @default true
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       privateEmptyConstructor: true
+   *
+   * ```
+   */
+
+  privateEmptyConstructor?: boolean;
+
+  /**
+   * @name unionKey
+   * @description specify the key to be used for Freezed union/sealed classes
+   * @default null
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       unionKey: 'type'
+   *
+   * ```
+   */
+
+  unionKey?: string;
+
+  /**
+   * @name unionValueCase
+   * @description specify the casing style to be used for Freezed union/sealed classes
+   * @default null
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       unionValueCase: 'FreezedUnionCase.pascal'
+   *
+   * ```
+   */
+
+  unionValueCase?: 'FreezedUnionCase.camel' | 'FreezedUnionCase.pascal';
+}
+
+export interface FieldConfig {
+  // TODO: apply it on the parameter block
+
+  /** marks a field as final */
+
+  final?: boolean;
+
+  /** marks a field as deprecated */
+
+  deprecated?: boolean;
+
+  /** annotate a field with a @Default(value: defaultValue) decorator */
+
+  defaultValue?: any;
+  /**
+   * @name customDecorators
+   * @description specific directives to apply to the field. All `mapsToFreezedAs` values except `custom` are parsed so use the name of the directive without the `@` symbol as the key of the customDecorators. With the `custom` value, whatever you use as the key of the custom directive is used just as it is, and the arguments spread into a parenthesis ()
+   * @default null
+   * @exampleMarkdown
+   * ```yml
+   * customDecorators: {
+   *    'default' : {
+   *        mapsToFreezedAs: '@Default',
+   *        arguments: ['$0'],
+   *      },
+   *      'deprecated' : {
+   *          mapsToFreezedAs: '@deprecated',
+   *       },
+   *      'readonly' : {
+   *          mapsToFreezedAs: 'final',
+   *       },
+   *      '@HiveField' : {
+   *          mapsToFreezedAs: 'custom',
+   *          arguments: ['1'],
+   *       }, # custom are used just as it given
+   * }
+   * ```
+   */
+
+  customDecorators?: CustomDecorator; // TODO:
+}
+
+export interface TypeSpecificFreezedConfig {
+  /** marks a type as deprecated */
+
+  deprecated?: boolean; // TODO: apply on the class and factory block
+
+  /** overrides the `globalFreezedConfig` for this type */
+
+  config?: FreezedConfig;
+
+  /** configure fields for this type */
+  fields?: Record<string, FieldConfig>; // TODO: apply on the class and factory block
+}
+
+export interface FreezedPluginConfig /* extends TypeScriptPluginConfig */ {
```

Comment:
```suggestion
export interface FlutterFreezedPluginConfig /* extends TypeScriptPluginConfig */ {
```

Config types always carry the full context, for example:

https://github.com/dotansimha/graphql-code-generator/blob/4b5036e4daaa3271f81b83ead857dcd892387714/packages/plugins/typescript/react-apollo/src/config.ts#L8

Reply:
Renamed as requested